src/test/capi_predict_validation_test.cpp

```diff
@@ -0,0 +1,725 @@
+}
+
+TEST_F(CAPIPredictValidationDynamicModel, RequestDimensionNotInRangeSecondPosition) {
+    //prepareKFSInferInputTensor(request, "Input_FP32_any_224:512_224:512_3_NHWC", {{1, 223, 224, 3}, "FP32"});
+    preparePredictRequest(request,
+        {
+            {"Input_FP32_any_224:512_224:512_3_NHWC",
+                std::tuple<ovms::shape_t, ovms::Precision>{{1, 223, 224, 3}, ovms::Precision::FP32}},
+            {"Input_U8_100:200_any_CN",
+                std::tuple<ovms::shape_t, ovms::Precision>{{101, 16}, ovms::Precision::U8}},
+        });
+    auto status = instance->mockValidate(&request);
+    EXPECT_EQ(status, ovms::StatusCode::INVALID_SHAPE) << status.string();
+}
+
+TEST_F(CAPIPredictValidationDynamicModel, RequestDimensionInRangeWrongTensorContent) {
+    //findKFSInferInputTensorContentInRawInputs(request, "Input_U8_100:200_any_CN")->clear();
+    badSize = true;
+    preparePredictRequest(request,
+        {
+            {"Input_FP32_any_224:512_224:512_3_NHWC",
+                std::tuple<ovms::shape_t, ovms::Precision>{{16, 300, 320, 3}, ovms::Precision::FP32}},
+            {"Input_U8_100:200_any_CN",
+                std::tuple<ovms::shape_t, ovms::Precision>{{101, 16}, ovms::Precision::U8}},
+        },
+        {}, putBufferInInputTensorContent, badSize);
+    auto status = instance->mockValidate(&request);
+    EXPECT_EQ(status, ovms::StatusCode::INVALID_CONTENT_SIZE) << status.string();
+}
+
+class CAPIPredictValidationPrecision : public ::testing::TestWithParam<ovms::Precision> {
+protected:
+    void SetUp() override {
+        auto precision = ovms::Precision::FP32;
+        mockedInputsInfo[tensorName] = std::make_shared<ovms::TensorInfo>(tensorName, precision, ovms::shape_t{1, DUMMY_MODEL_INPUT_SIZE}, ovms::Layout{"NC"});
+    }
+    ovms::InferenceRequest request{"model_name", 1};
+    ;
+    const char* tensorName = DUMMY_MODEL_INPUT_NAME;
+    ovms::tensor_map_t mockedInputsInfo;
+};
+
+TEST_P(CAPIPredictValidationPrecision, ValidPrecisions) {
+    ovms::Precision testedPrecision = GetParam();
+    mockedInputsInfo[tensorName]->setPrecision(testedPrecision);
+    preparePredictRequest(request,
+        {
+            {tensorName,
+                std::tuple<ovms::shape_t, ovms::Precision>{{1, DUMMY_MODEL_INPUT_SIZE}, testedPrecision}},
+        });
+    auto status = ovms::request_validation_utils::validate(request, mockedInputsInfo, "dummy", ovms::model_version_t{1});
+    EXPECT_EQ(status, ovms::StatusCode::OK) << "Precision validation failed:"
+                                            << toString(testedPrecision)
+                                            << " should pass validation";
+}
+
+INSTANTIATE_TEST_SUITE_P(
+    Test,
+    CAPIPredictValidationPrecision,
+    ::testing::ValuesIn(SUPPORTED_CAPI_INPUT_PRECISIONS),
+    [](const ::testing::TestParamInfo<CAPIPredictValidationPrecision::ParamType>& info) {
+        return toString(info.param);
+    });
+
```

Comment:
We miss tests for incorrect buffer type. For now we probably should handle CPU buffers anyway, but it is possible to pass any value as input buffer type. The same is true for unsupported CAPI precisions and any arbitrary value that could be passed in OVMS_InferenceRequestAddInput.

Reply:
Done.